homeassistant/components/matrix/__init__.py

```diff
@@ -444,16 +452,57 @@ async def _handle_multi_room_send(
             )
         await asyncio.wait(_tasks)
 
-    async def _send_image(
-        self, image_path: str, target_rooms: Sequence[RoomAnyID]
+    async def _send_image_url(
+        self, image_url: str, target_rooms: Sequence[RoomAnyID], verify_ssl: bool
     ) -> None:
-        """Upload an image, then send it to all target_rooms."""
-        _is_allowed_path = await self.hass.async_add_executor_job(
-            self.hass.config.is_allowed_path, image_path
+        """Upload an image from a URL, then send it to all target_rooms."""
+        # Get required image metadata.
+        if not self.hass.config.is_allowed_external_url(image_url):
+            _LOGGER.error("URL '%s' not in allow list", image_url)
+            return
+        session = aiohttp_client.async_get_clientsession(self.hass)
+        resp = await session.get(
+            image_url, raise_for_status=True, verify_ssl=verify_ssl
+        )
```

Comment:
I'm wondering if we want to let the potential exception here (and have a potentially "ugly" error in the logs), vs catching it and wrapping it slightly with "Oops, couldn't get this image to send it" (and still including the stack trace via `_LOGGER.exception`